exammode/views.py

```diff
@@ -0,0 +1,164 @@
+    access_mode = ACCESS.TEACHER
+
+    def get_context_data(self, **kwargs):
+        # Call the base implementation first to get the context
+        context = super(ExamReportView, self).get_context_data(**kwargs)
+        print(self)
+        # Create any data and add it to the context
+        context['active_exams'] = ExamSession.active_exams.is_active()
+        return context
+
+
+class ExamSessionEdit(UpdateView):
+    template_name = 'exammode/staff/edit_session_form.html'
+    form_class = ExamSessionForm
+    #fields = ['exam_module', 'can_start', 'duration', 'room']
+    model = ExamSession
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['course_instance'] = self.get_object().course_instance
+        return kwargs
+
+    '''
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        instance = self.get_object().course_instance
+
+        self.fields['course_instance'].queryset = CourseInstance.objects.filter(
+            id=instance.id)
+        self.fields['exam_module'].queryset = CourseModule.objects.filter(
+            course_instance=instance)
+
+    '''
+
+    def get_context_data(self, **kwargs):
+        context = super(ExamSessionEdit, self).get_context_data(**kwargs)
+        context['course_instance'] = self.get_object(
+        ).course_instance
+        return context
+
+    def get_success_url(self):
+        redirect_kwargs = {
+            'course_slug': self.kwargs['course_slug'],
+            'instance_slug': self.kwargs['instance_slug']
+        }
+        return reverse('exam-management', kwargs=redirect_kwargs)
+
+
+class ExamSessionDelete(DeleteView):
+    model = ExamSession
+
+    def get_success_url(self):
+        redirect_kwargs = {
+            'course_slug': self.kwargs['course_slug'],
+            'instance_slug': self.kwargs['instance_slug']
+        }
+        return reverse('exam-management', kwargs=redirect_kwargs)
+
+    def post(self, request, *args, **kwargs):
+        if "cancel" in request.POST:
+            url = self.get_success_url()
+            return redirect(url)
+        if "confirm" in request.POST:
+            return super(ExamSessionDelete, self).post(request, *args, **kwargs)
+
+
+class ExamManagementView(CourseInstanceMixin, BaseFormView):
+    access_mode = ACCESS.TEACHER
+    form_class = ExamSessionForm
+    template_name = "exammode/staff/exam_management.html"
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['course_instance'] = self.instance
+        return kwargs
+
+    def get_common_objects(self):
+        super().get_common_objects()
+        self.exam_sessions = list(self.exam_sessions)
+        self.note('exam_sessions')
+
+    def get_success_url(self):
+        return self.request.path_info
+
+    def form_valid(self, form):
+        form.save()
+        messages.success(self.request, _("Changes were saved succesfully."))
```

Comment:
Names `_` and `messages` are not defined. They could be imported in the beginning of the file:
 ```
from django.contrib import messages
from django.utils.translation import ugettext_lazy as _